pycroft/lib/finance/retransfer.py

```diff
@@ -0,0 +1,47 @@
+from collections.abc import Sequence
+from datetime import datetime, timedelta
+
+from sepaxml import SepaTransfer
+from sqlalchemy import select
+from sqlalchemy.orm import joinedload
+
+from pycroft import config
+from pycroft.helpers.utc import ensure_tz
+from pycroft.model import session
+from pycroft.model.finance import BankAccountActivity
+
+
+def get_activities_to_return() -> Sequence[BankAccountActivity]:
+    statement = (
+        select(BankAccountActivity)
+        .options(joinedload(BankAccountActivity.bank_account))
+        .filter(BankAccountActivity.transaction_id.is_(None))
+        .filter(BankAccountActivity.amount > 0)
+        .filter(BankAccountActivity.imported_at < ensure_tz(datetime.utcnow() - timedelta(days=14)))
+    )
+
+    return session.session.scalars(statement).all()
+
+
+def generate_activities_return_sepaxml(activities: list[BankAccountActivity]) -> bytes:
```

Comment:
This method needs a test – which is easy because you separated concerns well: this function has no implicit state nor side effects.

In this case I would actually just save the whole XML response in a test file, verify it visually, and access it via [`pkg_resources`](https://setuptools.pypa.io/en/latest/pkg_resources.html#basic-resource-access).